src/analysis/retail/rogue/outlaw/modules/apl/AplCheck.tsx

```diff
@@ -0,0 +1,268 @@
+import SPELLS from 'common/SPELLS';
+import { suggestion } from 'parser/core/Analyzer';
+import aplCheck, {
+  Apl,
+  build,
+  CheckResult,
+  PlayerInfo,
+  Rule,
+  tenseAlt,
+} from 'parser/shared/metrics/apl';
+import annotateTimeline from 'parser/shared/metrics/apl/annotate';
+import TALENTS from 'common/TALENTS/rogue';
+import {
+  and,
+  buffMissing,
+  buffPresent,
+  debuffMissing,
+  hasResource,
+  or,
+  describe,
+  buffStacks,
+  not,
+  optionalRule,
+} from 'parser/shared/metrics/apl/conditions';
+
+import { AnyEvent } from 'parser/core/Events';
+import RESOURCE_TYPES from 'game/RESOURCE_TYPES';
+import { SpellLink } from 'interface';
+import { ROLL_THE_BONES_BUFFS } from '../../constants';
+import { buffsCount } from './buffsCount';
+
+//--TODO: GS can't work for now until I finish working on gs debuff eventFabricator
+//        Add support for KiR builds
+//        BtE and SnD having variable duration depending on cp spent make their conditions innacurate for now
+//        1:43, 3:12, 3:44, 5:29 https://www.warcraftlogs.com/reports/t27zV8NLgPvJq3Kk#fight=17&type=damage-done&source=18, user cast rtb and the
+//            cast is flagged, this is due to my buffsCount function condition not having a slight delay akin to buff/debuffMissing
+//        Might wanna allow users to delay vanish without opp as it is barely worse
+//        SnD condition being rarely met it sometimes don't show in the apl if the user never casted the spell
+//        Add an optional rule to allow use of Ambush at 6cp with 2p and if bte isnt going to be pressed, not sure if fully possible (0.1% minmax)
+
+const hasFinisherCondition = () => {
+  //             this should be using: finishers.recommendedFinisherPoints()
+  return describe(hasResource(RESOURCE_TYPES.COMBO_POINTS, { atLeast: 6 }), (tense) => (
+    <>the finisher condition {tenseAlt(tense, 'is', 'was')} met</>
+  ));
+};
+
+//  /!\ Not sure this is working
+//  minEnergyThreshold will be the hard enforced energy rule while maxEnergyThreshold is the leeway accorded to the user
+const energyCondition = (minEnergyThreshold: number, maxEnergyThreshold: number) => {
+  return or(
+    hasResource(RESOURCE_TYPES.ENERGY, { atMost: minEnergyThreshold }),
+    optionalRule(hasResource(RESOURCE_TYPES.ENERGY, { atMost: maxEnergyThreshold })),
+  );
+};
+
+//  if snc is down and rtbBuffCount < 2 should reroll, we are allowing the user to also keep a single Broadside as it is barely worse
+const rtbCondition = () => {
+  const rtbBuffsToCheck = ROLL_THE_BONES_BUFFS.filter((spell) => spell !== SPELLS.GRAND_MELEE);
+  return and(
+    describe(
+      or(
+        // allow rerolling if you're missing either SnC or BT, but don't require it
+        optionalRule(or(buffMissing(SPELLS.SKULL_AND_CROSSBONES), buffMissing(SPELLS.BROADSIDE))),
+        // require rerolling if you're missing both SnC and BT
+        and(buffMissing(SPELLS.SKULL_AND_CROSSBONES), buffMissing(SPELLS.BROADSIDE)),
+      ),
```

Comment:
i *think* this should work because the `or` will prevent the infectious interaction of `optionalRule` with `and`? something to keep an eye on.